config/v2/eigendaflags/cli.go

```diff
@@ -168,12 +194,20 @@ func ReadClientConfigV2(ctx *cli.Context) (common.ClientConfigV2, error) {
 	}
 
 	return common.ClientConfigV2{
-		DisperserClientCfg:         disperserConfig,
-		PayloadDisperserCfg:        readPayloadDisperserCfg(ctx),
-		RelayPayloadRetrieverCfg:   readRetrievalConfig(ctx),
-		PutTries:                   ctx.Int(PutRetriesFlagName),
-		MaxBlobSizeBytes:           maxBlobLengthBytes,
-		EigenDACertVerifierAddress: ctx.String(CertVerifierAddrFlagName),
+		DisperserClientCfg:            disperserConfig,
+		PayloadDisperserCfg:           readPayloadDisperserCfg(ctx),
+		RelayPayloadRetrieverCfg:      readRelayRetrievalConfig(ctx),
+		ValidatorPayloadRetrieverCfg:  readValidatorRetrievalConfig(ctx),
+		PutTries:                      ctx.Int(PutRetriesFlagName),
+		MaxBlobSizeBytes:              maxBlobLengthBytes,
+		EigenDACertVerifierAddress:    ctx.String(CertVerifierAddrFlagName),
+		BLSOperatorStateRetrieverAddr: ctx.String(BLSOperatorStateRetrieverFlagName),
+		EigenDAServiceManagerAddr:     ctx.String(ServiceManagerAddrFlagName),
+		// we don't expose this configuration to users, as all production use cases should have
+		// both retrieval methods enabled. This could be exposed in the future, if necessary.
+		// Note the order of these retrievers, which is significant: the relay retriever will be
+		// tried first, and the validator retriever will only be tried if the relay retriever fails
```

Comment:
Is this ordering thing true? In the store_manager_builder, you build them always relay first, regardless of config ordering.

Reply:
Yes, the ordering thing is true. Relevant here isn't construction order: rather, it's the slice iteration that takes place in the `Get` pathway

```
	for _, retriever := range e.retrievers {
		payload, err := retriever.GetPayload(ctx, &cert)
		if err == nil {
			return payload.Serialize(), nil
		}

		e.log.Debugf("Payload retriever failed: %v", err)
	}
```

https://github.com/Layr-Labs/eigenda-proxy/blob/4d8097dac6d8442705ba2459e6fd38670d254869/store/generated_key/v2/eigenda.go#L69

Reply:
But isnt the ordering of e.retrievers populated by the build step, which changes the ordering? https://github.com/Layr-Labs/eigenda-proxy/blob/4d8097dac6d8442705ba2459e6fd38670d254869/store/storage_manager_builder.go#L262-L272

Reply:
You're 💯% correct, I didn't read my own code carefully enough...

bug fixed in [6b3f050](https://github.com/Layr-Labs/eigenda-proxy/pull/369/commits/6b3f0508a915e6605b141c8432dde04867e938b4)